common/web/types/src/kmx/kmx-plus-builder/build-elem.ts

```diff
@@ -1,16 +1,19 @@
 import { constants } from "@keymanapp/ldml-keyboard-constants";
 import { ElementString } from "../element-string.js";
 import { Elem } from "../kmx-plus.js";
-import { build_strs_index, BUILDER_STRS } from "./build-strs.js";
+import { build_strs_index, BUILDER_STR_REF, BUILDER_STRS } from "./build-strs.js";
 import { BUILDER_SECTION } from "./builder-section.js";
-import { build_uset_index, BUILDER_USET } from "./build-uset.js";
+import { build_uset_index, BUILDER_USET, BUILDER_USET_REF } from "./build-uset.js";
 
 /* ------------------------------------------------------------------
  * elem section
    ------------------------------------------------------------------ */
 
+/** return from build_elem_index */
+export type BUILDER_ELEM_REF = number;
+
 interface BUILDER_ELEM_ELEMENT {
-  element: number;  // str | UTF-32 char
+  element: BUILDER_STR_REF | BUILDER_USET_REF | number;  // str | UTF-32 char
```

Comment:
I wonder if we should have a `BUILDER_UTF32_CHAR` type?

Reply:
Or perhaps `BUILDER_U32CHAR`

Reply:
Me:  It could be a string! it could be a UnicodeSet! it could be a 32-bit char!

Typescript: It's a number.
<img width="724" alt="image" src="https://github.com/keymanapp/keyman/assets/855219/c0e78986-7fb0-4d93-924b-c20f119bf3c2">